spec.html

```diff
@@ -9948,91 +9948,58 @@ <h1>Static Semantics: StringValue</h1>
       </emu-clause>
     </emu-clause>
 
-    <emu-clause id="sec-reserved-words">
-      <h1>Reserved Words</h1>
-      <p>A reserved word is an |IdentifierName| that cannot be used as an |Identifier|.</p>
+    <emu-clause id="sec-keywords-and-reserved-words" oldids="sec-reserved-words,sec-keywords,sec-future-reserved-words">
+      <h1>Keywords and Reserved Words</h1>
+      <p>A <em>keyword</em> is a token that matches |IdentifierName|, but also has a syntactic use; that is, it appears literally, in a `fixed width` font, in some syntactic production. The keywords of ECMAScript include `if`, `while`, `async`, `await`, and many others.</p>
+      <p>A <em>reserved word</em> is an |IdentifierName| that cannot be used as an identifier. Many keywords are reserved words, but some are not, and some are reserved only in certain contexts. `if` and `while` are reserved words. `await` is reserved only inside async functions and modules. `async` is not reserved; it can be used as a variable name or statement label without restriction.</p>
+      <p>This specification uses a combination of grammatical productions and early error rules to specify which names are valid identifiers and which are reserved words. All tokens in the |ReservedWord| list below, except for `await` and `yield`, are unconditionally reserved. Exceptions for `await` and `yield` are specified in <emu-xref href="#sec-identifiers"></emu-xref>, using parameterized syntactic productions. Lastly, several early error rules restrict the set of valid identifiers. See <emu-xref href="#sec-identifiers-static-semantics-early-errors"></emu-xref>, <emu-xref href="#sec-let-and-const-declarations-static-semantics-early-errors"></emu-xref>, <emu-xref href="#sec-for-in-and-for-of-statements-static-semantics-early-errors"></emu-xref>, and <emu-xref href="#sec-class-definitions-static-semantics-early-errors"></emu-xref>. In summary, there are five categories of identifier names:</p>
+      <ul>
+        <li>
+          <p>Those that are always recognized as identifiers, such as `Math`, `window`, `toString`, and `_`;</p>
```

Comment:
Something I forgot to mention: If you took this item and moved it to the last or second-last spot, then the list would have a more logical order: increasing number of contexts in which the names in question are recognized as identifiers. (None, a few, almost all, all.)

The last set (names that are always allowed as identifier but also appear as keywords) is necessarily a subset of the first (names that are always allowed as identifiers), so it would be good to have those at least adjacent, if not combined.

Reply:
I could go either way on the list order, but I did think about it and decide to stick with Allen's order. I think it is friendlier to a person reading the list for the first time from the top down. First they are given the two biggest categories, which are the endpoints of the range of identifier-ness; then the three kinds of special conditional cases.

The sets are meant to be disjoint. The wording "always recognized as identifiers" is doing some subtle work in this regard, distinguishing those from the last set (always *allowed* as identifiers, but sometimes recognized as keywords). I could change it to "Those that are always allowed as identifiers, and are not keywords, such as...". Better?

Reply:
> I could change it to "Those that are always allowed as identifiers, and are not keywords, such as...". Better?

No, because they _are_ keywords as defined in the first sentence of this section:

> A keyword is a token that matches IdentifierName, but also has a syntactic use; that is, it appears literally, in a fixed width font, in some syntactic production. 

but perhaps the last category could be a bit more explicit:

*  Those that are always allowed as identifiers, but also appear as keywords within certain syntactic productions at places where _Identifer_ is not allowed: ...

Reply:
> I could go either way on the list order, but I did think about it [...]

Okay.

> The sets are meant to be disjoint. The wording "always recognized as identifiers" is doing some subtle work in this regard, distinguishing those from the last set (always _allowed_ as identifiers, but sometimes recognized as keywords).

Ah, I see, hm. Then it seems like the other items should also talk about the keyword-iness of their members.

> I could change it to "Those that are always allowed as identifiers, and are not keywords, such as...". Better?

That's for the *first* bullet, right? Yeah, I think that would be better.

Reply:
Yes, I meant the first bullet, sorry for the confusion. I'll change it.

> Then it seems like the other items should also talk about the keyword-iness of their members.

I wanted to avoid the wrinkle that future reserved words aren't keywords (yet).

Reply:
>     * Those that are always allowed as identifiers, but also appear as keywords within certain syntactic productions at places where _Identifer_ is not allowed: ...

Great, I'm adding this too.

Reply:
Is this resolved now?